src/lang-promql/complete/index.ts

```diff
@@ -32,19 +34,20 @@ export interface CompleteStrategy {
 
 // CompleteConfiguration should be used to customize the autocompletion.
 export interface CompleteConfiguration {
-  // Provide these settings when not using a custom PrometheusClient.
-  url?: string;
-  lookbackInterval?: number;
-  httpErrorHandler?: (error: any) => void;
-  fetchFn?: FetchFn;
-
+  remote: {
+    // Provide these settings when not using a custom PrometheusClient.
+    url?: string;
+    lookbackInterval?: number;
+    httpErrorHandler?: (error: any) => void;
+    fetchFn?: FetchFn;
+    // When providing this custom PrometheusClient, the settings above will not be used.
+    prometheusClient?: PrometheusClient;
```

Comment:
Doesn't have to be in this PR, but would it now make more sense to package the `url` ... `fetchFn` fields into their own `PrometheusConfig`, and then just define the `remote` field as: `remote: PrometheusConfig | PrometheusClient`? Or something like that. Then TypeScript will prevent invalid configuration combinations and things would be clearer.

Reply:
I won't mind to do it, I would just be sure, it won't be annoying to use. I mean if I want to change just the url, then I have to create the struct:

```typescript
{
  remote: {
    PrometheusConfig: {
      url: 'http://prometheus.land'
    }
  }
}
```
Don't know if it will bother developer to have so much different interface, just to configure an url.
What do you think ?

Reply:
The middle step would be left out if you do it like I suggest above:

```typescript
{
  remote: {
    url: 'http://prometheus.land'
  }
}
```

TypeScript would automatically notice that I'm now passing in something that satisfies the interface `PrometheusConfig`, not `PrometheusClient`. Of course then the `url` field should be marked mandatory in `PrometheusConfig`.

Reply:
ah yeah ok, I see what you meant. Nice thanks, will do that

Reply:
done